src/master-playlist-controller.js

```diff
@@ -211,9 +211,23 @@ export class MasterPlaylistController extends videojs.EventTarget {
     this.mode_ = mode;
     this.useCueTags_ = useCueTags;
     if (this.useCueTags_) {
-      this.cueTagsTrack_ = this.tech_.addTextTrack('metadata',
-        'ad-cues');
-      this.cueTagsTrack_.inBandMetadataTrackDispatchType = '';
+      let track;
+      const tracks = this.tech_.textTracks();
+      let i = tracks.length;
+
+      // check if an track already exists for id3 cue tags
+      while (i-- && !track) {
```

Comment:
Might just be preference, but any specific reason for counting backwards with the while instead of using `for (let i = 0; i < tracks.length; i++)`?

Might also be good to have a separate function for some of this code, just to keep the constructor cleaner.